edgedb/protocol/protocol.pyx

```diff
@@ -249,12 +234,10 @@ cdef class SansIOProtocol:
         buf.write_byte(output_format)
         buf.write_byte(CARDINALITY_ONE if expect_one else CARDINALITY_MANY)
         buf.write_len_prefixed_utf8(query)
-        if self.state is not None:
-            buf.write_bytes(self.state_type_id)
-            buf.write_buffer(self.state)
-        else:
-            buf.write_bytes(NULL_CODEC_ID)
```

Comment:
I'm not sure this change is sound.  If the client has no state to send, then sending `NULL` is preferable, because then this reduces the risk of concurrent DDL crashing queries, which affects our test suite, see https://github.com/edgedb/edgedb/runs/7141764907?check_suite_focus=true#step:17:605

Reply:
Hm, nevermind `encode_state` seems to do this.